src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -560,11 +563,7 @@ private bool ValidatePackagesSha512(PackagesLockFile lockFile, LockFile assetsFi
 
             if (_request.ProjectStyle == ProjectStyle.DotnetCliTool)
             {
```

Comment:
I know that this is old code, but it would be great if we could add comments why modify the request based on the ProjectStyle.

Reply:
@nkolev92 can you remind me why this condition is here so I can add a comment?

Reply:
DotnetCliTool restores are special because the the assets file location is not known until after the restore itself. 
So we just clean up.